src/clients/memcache.rs

```diff
@@ -0,0 +1,297 @@
+use super::*;
+use crate::net::Connector;
+use protocol_memcache::{Compose, Parse, Request, Response, Ttl};
+use session::{Buf, BufMut, Buffer};
+use std::borrow::{Borrow, BorrowMut};
+
+/// Launch tasks with one conncetion per task as memcache protocol is not mux-enabled.
+pub fn launch_tasks(runtime: &mut Runtime, config: Config, work_receiver: Receiver<WorkItem>) {
+    debug!("launching memcache protocol tasks");
+
+    // create one task per connection
+    for _ in 0..config.client().unwrap().poolsize() {
+        for endpoint in config.target().endpoints() {
+            runtime.spawn(task(
+                work_receiver.clone(),
+                endpoint.clone(),
+                config.clone(),
+            ));
+        }
+    }
+}
+
+#[allow(clippy::slow_vector_initialization)]
+async fn task(work_receiver: Receiver<WorkItem>, endpoint: String, config: Config) -> Result<()> {
+    let connector = Connector::new(&config)?;
+
+    let mut stream = None;
+    let parser = protocol_memcache::ResponseParser {};
+    let mut read_buffer = Buffer::new(4096);
```

Comment:
This needs to be configurable? If we send very large payload we don't want either allocation or partial read on the client side to affect latency reading.